storage/remote/queue_manager.go

```diff
@@ -375,12 +398,46 @@ func (t *QueueManager) Start() {
 	go t.reshardLoop()
 }
 
+// loadCheck will check if the SegmentRecord.json contents and segment is valid.
+func (t *QueueManager) loadCheck() bool {
+	var check CheckpointRecord
+
+	jsonFile, err := os.Open("data/CheckpointRecord.json")
+	defer jsonFile.Close()
+
+	byteValue, _ := ioutil.ReadAll(jsonFile)
+	err = json.Unmarshal(byteValue, &check)
+
+	if err != nil {
+		level.Error(t.logger).Log("err", err)
+		return false
+	}
+
+	for _, record := range check.Checkpoints {
+		segmentNumber, err := strconv.Atoi(record.Segment)
+
+		if err != nil {
+			level.Error(t.logger).Log("err", err)
+			return false
+		}
+
+		// Segmentnumber out of bounds
+		if segmentNumber < wal.MinSegmentNumber || segmentNumber > wal.MaxSegmentNumber {
```

Comment:
We should maybe have the watcher have a `ValidSegment` function that uses `wal.Segments` to check if `segmentNumber` exists in the current WAL directory. Thoughts @csmarchbanks ?